src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,57 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int SUPPLY_TYPE = 0;
+    private static final int BUY_TYPE = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = readFile(fromFileName);
+        String report = createReport(dataFromFile);
+        writeToFile(report, toFileName);
+    }
+
+    private String readFile(String fileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            StringBuilder data = new StringBuilder();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                data.append(line).append("\n");
+            }
+            return data.toString();
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from the file", e);
+        }
+    }
+
+    private String createReport(String data) {
+        int supply = 0;
+        int buy = 0;
+        String[] lines = data.split("\n");
+        for (String line : lines) {
+            String[] parts = line.split(",");
+            int operationType = "supply".equals(parts[0]) ? SUPPLY_TYPE : BUY_TYPE;
+            int amount = Integer.parseInt(parts[1]);
```

Comment:
avoid using magic numbers

all indexes should be constant with a self-descriptive message.